### PR TITLE
added action:resumed and action:resume + code cleanup

### LINKED ISSUE #1889: error in parseACK.c for infoSettings.machine_size_max ?
//----------------------------------------
      // Tuning parsed responses
      //----------------------------------------

      // parse and store build volume size
      else if (ack_seen("work:"))
      {
        if (ack_seen("min:"))
        {
          if (ack_seen("X:")) infoSettings.machine_size_min[X_AXIS] = ack_value();
          if (ack_seen("Y:")) infoSettings.machine_size_min[Y_AXIS] = ack_value();
          if (ack_seen("Z:")) infoSettings.machine_size_min[Z_AXIS] = ack_value();
        }
        if (ack_seen("max:"))
        {
          if (ack_seen("X:")) infoSettings.machine_size_min[X_AXIS] = ack_value();     <--- is not infoSettings.machine_size_max
          if (ack_seen("Y:")) infoSettings.machine_size_min[Y_AXIS] = ack_value();     <--- is not infoSettings.machine_size_max
          if (ack_seen("Z:")) infoSettings.machine_size_min[Z_AXIS] = ack_value();     <--- is not infoSettings.machine_size_max
        }
      }

### LINKED ISSUE #1895: [BUG] after loading new firmware on tft 35 the led green is with wite color!and the others with the white color ((( if i switch to the old version everything is good)
### Steps to reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior**

**Actual behavior**

### Additional Information
* Include a ZIP file containing your `Configuration.h` or use [Pastebin](https://pastebin.com/) and paste a link in this issue.
* Provide pictures or links to videos that clearly demonstrate the issue.

### PR DESCRIPTION
**Pull request by digant73**
**IMPROVEMENTS:**
 * added "action:resumed" and "action:resume" on parseAck fixing a missing dialog for the M600 not properly working reported in #1829. Now, resuming a print, the "Resume" button is properly switched to "Pause" button even in case REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER or ADVANCED_PAUSE_FEATURE is not enabled in Marlin
* Code cleanup. Reviewed .h and .c files just to include only the needed dependency files (in particular avoiding to include general files in the .h files)
* added ENABLE_MBL_START_Z in Configuration.h. It enables/disables "level_z_pos" configurable parameter value as the starting Z height for each point during MBL process.
* minor cleanup

**FIXES:**
fixes #1889
fixes #1895

**PR STATE:** ready for merging

### CODE REVIEW

**Review by sarvenn — APPROVED** [review]

File: TFT/src/User/API/menu.h [hunk 1]
```diff
@@ -5,8 +5,8 @@
 extern "C" {
 #endif
 
-#include <stdint.h>
 #include <stdbool.h>
+#include <stdint.h>
```
Comment (guruathwal) [inline TFT/src/User/API/menu.h:9]: is there any special reason for this change?

Comment (digant73) [reply TFT/src/User/API/menu.h:9]: It seems only a wrong view update on git. the last pushed menu.h is as you suggested. thanks anyway

File: TFT/src/User/Menu/MarlinMode.c [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #include "MarlinMode.h"
-#include "spi_slave.h"
-#include "HD44780.h"
 #include "includes.h"
+#include "HD44780.h"
+#include "spi_slave.h"
```
Comment (guruathwal) [inline TFT/src/User/Menu/MarlinMode.c:4]: spi_slave.h should be included first.

Comment (digant73) [reply TFT/src/User/Menu/MarlinMode.c:4]: ok

**Comment by sarvenn** [comment]
@digant73 hi,
Is there any possibility to include the level_z parameter to effect or not to effect MBL Z initial height and use marlin's setting in configuration.h.
I know that you included in TFT's configuration.h file but to be more easier usage, could it be integrated to config.ini ?
Would it hurt the RAM usage? If so, yes I may prefer it to change in configuration.h and compile my own fw for my TFT and not to use pre-compiled files.

**Comment by digant73** [comment]
it will affect FLASH and RAM. For that reason it was made configurable only on Configuration.h

**Comment by sarvenn** [comment]
great job, thnx indeed.

**Pull request merged**